File: clang/test/CIR/CodeGen/OpenMP/taskwait.cpp
// RUN: %clang_cc1 -std=c++20 -triple x86_64-unknown-linux-gnu -fopenmp-enable-irbuilder -fopenmp -fclangir -emit-cir %s -o %t.cir
// RUN: FileCheck --input-file=%t.cir %s
// XFAIL: *

// CHECK: cir.func
void omp_taskwait_1(){
// CHECK: omp.taskwait
  #pragma omp taskwait
}

File: flang/test/Lower/OpenMP/taskwait.f90
!RUN: %flang_fc1 -emit-hlfir -fopenmp %s -o - | FileCheck %s

!CHECK-LABEL: @_QPomp_taskwait
subroutine omp_taskwait
  !CHECK: omp.taskwait
  !$omp taskwait
  !CHECK: fir.call @_QPfoo() {{.*}}: () -> ()
  call foo()
  !CHECK: omp.taskwait
  !$omp taskwait
end subroutine omp_taskwait
